### PR TITLE
Fix: make sure we ignore notifications for open rooms

### LINKED ISSUE #866: Notifications are displayed for messages sent to the current room
### Steps to reproduce

1. Open a room in the app.
2. From another client, send a message.

#### What did you expect?

Since the room is already displayed, no notification should be displayed.

#### What happened instead?

The notification is displayed.

### Your phone model

OnePlus 6T

### Operating system version

Android 11.0

### Application version and app store

Element X - develop (106d0708462a0f45a0c04102012f8c3cdbe6e1c3)

### Homeserver

_No response_

### Will you send logs?

No

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by jmartinesp**
## Changes

To ignore notifications for the currently displayed room, we were previously checking `ProcessLifecycleOwner.get().lifecycle` in `NotifiableMessageEvent`. However, this value seems to always return the `INITIALIZED`, not ever updating.

The new changes add an observer that publishes the current app state to `AppNavigationStateService`.

## Why

Fixes #866.

### CODE REVIEW

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt [hunk 1]
```diff
@@ -87,7 +87,7 @@ class RustMatrixClient constructor(
 
     override val sessionId: UserId = UserId(client.userId())
     private val app = client.app().use { builder ->
-        builder.finish()
+        builder.withEncryptionSync(false, "").finish()
```
Comment (ganfra) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt:90]: Should be part of another PR if we really enable this I think

File: app/src/main/kotlin/io/element/android/x/ElementXApplication.kt [hunk 2]
```diff
@@ -43,6 +49,13 @@ class ElementXApplication : Application(), DaggerComponentOwner {
             initializeComponent(MatrixInitializer::class.java)
             initializeComponent(EmojiInitializer::class.java)
         }
+
+        val appNavigationStateService = appComponent.appNavigationStateService()
+        ProcessLifecycleOwner.get().lifecycle.addObserver(object : LifecycleEventObserver {
```
Comment (ganfra) [inline app/src/main/kotlin/io/element/android/x/ElementXApplication.kt:54]: Can you not listen to this directly in the `AppNavigationStateService`?

File: services/appnavstate/api/src/main/kotlin/io/element/android/services/appnavstate/api/AppNavigationStateService.kt [hunk 3]
```diff
@@ -24,6 +25,9 @@ import kotlinx.coroutines.flow.StateFlow
 
 interface AppNavigationStateService {
     val appNavigationStateFlow: StateFlow<AppNavigationState>
+    val appIsInForeground: StateFlow<Boolean>
```
Comment (ganfra) [inline services/appnavstate/api/src/main/kotlin/io/element/android/services/appnavstate/api/AppNavigationStateService.kt:28]: Maybe encapsulate so we only have one StateFlow?
Something like that : (not sure about the naming)
```
data class AppNavigationState(
    val isInForeground: Boolean,
    val currentNavigationStep: NavigationStep 
)
```

**Review by ganfra — COMMENTED** [review]
Some remarks

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt (hunk 1 above)
Comment (jmartinesp) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt:90]: My bad, I thought I had removed that!

**Review by csmith — APPROVED** [review]

**Comment by jmartinesp** [comment]
Just to double check @ganfra any further comments after the latest changes?

**Pull request merged**